tycho-indexer/src/extractor/dynamic_contract_indexer/cache.rs

```diff
@@ -204,6 +204,27 @@ where
         self.permanent.get(k)
     }
 
+    /// Retrieves all values for a key from all layers, starting from the latest pending layer.
+    /// Returns a vector of all values for the key, starting from the latest pending layer.
+    /// If the key is not found in any layer, returns None.
+    pub(super) fn get_all(&self, k: &K) -> Option<Vec<&V>> {
+        let mut res = Vec::new();
+        for layer in self.pending.iter().rev() {
+            if let Some(v) = layer.data.get(k) {
+                res.push(v);
+            }
+        }
+        if let Some(v) = self.permanent.get(k) {
+            res.push(v);
+        }
+
+        if res.is_empty() {
+            None
+        } else {
+            Some(res)
+        }
+    }
```

Comment:
I think this would be cleaner if you just returned an iterator here. 

Or alternatively the return collection could be generic too:

```
pub(super) fn get_all<C>(&self, k: &K) -> Option<C>
where
    C: FromIterator<&V>,
{
    let mut iter = self
        .pending
        .iter()
        .rev()
        .filter_map(|layer| layer.data.get(k))
        .chain(self.permanent.get(k));

    let res: C = iter.collect();

    if res.into_iter().next().is_none() {
        None
    } else {
        Some(res)
    }
}
```